pizza_cutter/des_pizza_cutter/_se_image.py

```diff
@@ -1486,15 +1517,139 @@ def resample(
         bmask[y_rs[~msk], x_rs[~msk]] = BMASK_EDGE
         resampled_data['bmask'] = bmask
 
-        # 4. do the nearest pixel for the pmask
+        # 5. do the nearest pixel for the pmask
         pmask = np.zeros((box_size, box_size), dtype=self.pmask.dtype)
         pmask[y_rs[msk], x_rs[msk]] = self.pmask[
             y_rs_se[msk], x_rs_se[msk]]
         pmask[y_rs[~msk], x_rs[~msk]] = BMASK_EDGE
         resampled_data['pmask'] = pmask
         resampled_data['pmask'][edge] |= BMASK_RESAMPLE_BOUNDS
 
-        # 5. do the PSF image
+        return resampled_data
+
+    def resample_psf(
+        self, *, wcs, wcs_position_offset, wcs_interp_shape,
+        psf_x_start, psf_y_start, psf_box_size,
+        se_wcs_interp_delta, coadd_wcs_interp_delta,
+        dx=0, dy=0,
+    ):
+        """Resample a SEImageSlice PSF to a new WCS.
+
+        The resampling is done as follows. For each pixel in the destination
+        image, we compute its location in the source image. Then we use a
+        Lanczos3 interpolant in the source grid to compute the value of the
+        pixel in the destination image.
+
+        NOTE: For destination grids that are far from the input grid, this
+        function will not work.
+
+        NOTE: Typically, the input WCS object will be for a coadd.
+
+        NOTE: The parameters dx, dy are intended for use in correction for mean
+        astrometric offsets due to color-dependent astrometry in coadd. The idea is
+        that we have some default WCS color that is used everywhere when building
+        coadds. We know this color is not correct for any given object which has some
+        other color and thus a different astrometry and different PSF. We can correct
+        for this by computing
+
+            1) the PSF at objects color but with the WCS of the default color.
+            2) the offset in SE pixels of a fixed location in ra,dec between the two
+               colors
+
+               dx, dy = (
+                   WCS^(-1)(ra, dec, object color)
+                   - WCS^(-1)(ra, dec, default color)
+               )
+
+        Then when we interpolate the SE PSF image to the coadd frame, we add -dx, -dy
+        to the locations of the SE pixels computed with the default color. For a
+        positive (dx, dy), this will shift the center of SE PSF by (dx, dy). Remember
+        we shift the coordinates where we evaluate the image before coadding by
+        -dx, -dy. The positive shift in the PSF center is the exact shift we
+        (incorrectly) applied to the image when coadding originally (where we also
+        evaluated the image at coordinates shifted by -dx, -dy). We now include it in
+        the PSF center so that when we analyze the data with the PSF at the object's
+        color, we get the right PSF size and extra braodening due to astrometric errors.
```

Comment:
@rmjarvis Please take a look at this if you can. The docstring works out the proper way to apply the astrometric shift when interpolating the PSF. Needs a double check (and lots of tests ofc!)

Reply:
So the main confusing bit here is of course what the right sign is.  I'm reasonably confident that the sign of things you lay out here is correct.  

I'm also reasonably confident that it won't matter much (maybe at all) in practice, since the extra PSF broadening you would get be swapping the sign of everything is basically the same.  In fact, I'm having trouble concocting a test that would be able to discern the difference in terms of a shape measurement.  So it's possible that it doesn't matter at all.  (For shapes -- if we cared about the astrometry of galaxies, that would show a difference.)